platform/cxxsupport/mstd_span

```diff
@@ -0,0 +1,381 @@
+    constexpr span<element_type, Count> last() const
+    {
+        static_assert(Count <= extent);
+        MBED_ASSERT(Count <= size());
+        return {data() + (size() - Count), Count};
+    }
+
+    template<size_t Offset, size_t Count = dynamic_extent>
+    constexpr span<element_type, Count != dynamic_extent ? Count
+        : (Extent != dynamic_extent ? Extent - Offset : dynamic_extent)> subspan() const
+    {
+        static_assert(Offset <= extent && (Count == dynamic_extent || Count <= extent - Offset));
+        // Only check against Offset == 0 to prevent a warning for subspan<0, N>
+        MBED_ASSERT((Offset == 0 || Offset <= size())
+                    && (Count == dynamic_extent || Count <= size() - Offset));
+        return {data() + Offset, Count != dynamic_extent ? Count : size() - Offset};
+    }
+
+    constexpr span<element_type, dynamic_extent> first(index_type count) const
+    {
+        MBED_ASSERT(count <= size());
+        return {data(), count};
+    }
+
+    constexpr span<element_type, dynamic_extent> last(index_type count) const
+    {
+        MBED_ASSERT(count <= size());
+        return {data() + (size() - count), count};
+    }
+
+    constexpr span<element_type, dynamic_extent>
+        subspan(index_type offset, index_type count = dynamic_extent) const
+    {
+        MBED_ASSERT(offset <= size() && (count == dynamic_extent || count <= size() - offset ));
+        return {data() + offset, count == dynamic_extent ? size() - offset : count };
+    }
+
+    // Observers
+    constexpr index_type size() const noexcept
+    {
+        return _storage._size;
+    }
+
+    constexpr index_type size_bytes() const noexcept
+    {
+        return size() * sizeof(element_type);
+    }
+
+    constexpr bool empty() const noexcept
+    {
+        return size() == 0;
+    }
+
+    // Element access
+    constexpr reference operator[](index_type idx) const
+    {
+        MBED_ASSERT(idx < size());
+        return *(data() + idx);
+    }
+
+    constexpr reference front() const
+    {
+        MBED_ASSERT(not empty());
+        return *data();
+    }
+
+    constexpr reference back() const
+    {
+        MBED_ASSERT(not empty());
+        return *(data() + (size() - 1));
+    }
+
+    constexpr pointer data() const noexcept
+    {
+        return _storage._data;
+    }
+
+    // Iterators
+    constexpr iterator begin() const noexcept
+    {
+        return data();
+    }
+
+    constexpr iterator end() const noexcept
+    {
+        return data() + size();
+    }
+
+    constexpr reverse_iterator rbegin() const noexcept
+    {
+        return reverse_iterator(end());
+    }
+
+    constexpr reverse_iterator rend() const noexcept
+    {
+        return reverse_iterator(begin());
+    }
+
+private:
+    detail::storage<element_type, extent> _storage;
+};
+
+template<class ElementType, size_t Extent>
+span<const unsigned char, Extent == dynamic_extent ? dynamic_extent : sizeof(ElementType) * Extent>
```

Comment:
Don't think we should do this - I want everything in mstd to be conformant if there.

`as_bytes` has to return a `byte_t` span.

I guess you could have `as_uint8_t`, as mbed APIs use `uint8_t` a lot (not `unsigned char`), but even then I've been loathe to add extensions inside `mstd`. Only doing the standard stuff and nothing more saves documentation :)

I could imagine maybe having an `mbed::as_uint8_t` somewhere else, and we could have `mbed::as_uint8_t<mstd::span>`?

But then, I guess we still have `make_span` like `Span` did as a substitute for the template deduction guides... So maybe this can stay here. But still, not under the `as_bytes` name.

Reply:
(We can't really provide `mstd::byte_t` now, as to make it usable it needs a relaxation to `enum class` in the core language - if I understand correctly.)